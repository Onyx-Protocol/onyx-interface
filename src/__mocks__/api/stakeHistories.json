{
  "data": [
    {
      "id": 291,
      "address": "0x0000000000000000000000000000000000000000",
      "tx_hash": "0xf97a7673403a643ad23edb9ffa4c60bf4192998aaf436782240b84be94d3aba1",
      "block_hash": "0x7d6991ef8f031e9cdc4ced2f924dee008e505c741f169101124aee16dad795f1",
      "type": 0,
      "amount": "11075000000000000000",
      "reward": "0",
      "price": "0.033695332045177356",
      "created_at": "2022-12-14T19:04:11.000Z",
      "updated_at": "2022-12-14T19:04:11.000Z"
    }
  ],
  "metadata": {
    "page": 1,
    "limit": 5,
    "totalItem": 1,
    "totalPage": 1
  }
}
