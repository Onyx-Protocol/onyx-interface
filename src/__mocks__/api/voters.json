{
  "data": [
    {
      "id": "40b3dcb3-f337-4713-ae4c-cae927351550",
      "address": "0xe864c9439eb042e969c59dd13d38cfdb28e16d73",
      "has_voted": 1,
      "support": 1,
      "votes": "232456299820100000000000000",
      "proposal_id": 6,
      "block_number": 14636926,
      "block_timestamp": 1650660975,
      "created_at": "2022-07-11T01:00:47.233Z",
      "updated_at": "2022-07-11T01:01:00.000Z"
    }
  ],
  "metadata": {
    "sumVotes": "232456299820100000000000000",
    "page": 1,
    "limit": 4,
    "totalItem": 1,
    "totalPage": 1
  }
}