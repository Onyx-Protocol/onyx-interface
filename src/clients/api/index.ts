export { default as queryClient } from './queryClient';

// Mutations
export { default as enterMarkets } from './mutations/enterMarkets';
export * from './mutations/enterMarkets';
export { default as useEnterMarkets } from './mutations/enterMarkets/useEnterMarkets';

export { default as exitMarket } from './mutations/exitMarket';
export * from './mutations/exitMarket';
export { default as useExitMarket } from './mutations/exitMarket/useExitMarket';

export { default as approveToken } from './mutations/approveToken';
export * from './mutations/approveToken';
export { default as useApproveToken } from './mutations/approveToken/useApproveToken';

export { default as approveNftForAll } from './mutations/approveNftForAll';
export * from './mutations/approveNftForAll';
export { default as useApproveNftForAll } from './mutations/approveNftForAll/useApproveNftForAll';

export { default as supplyNonEth } from './mutations/supplyNonEth';
export * from './mutations/supplyNonEth';
export { default as useSupplyNonEth } from './mutations/supplyNonEth/useSupplyNonEth';
export * from './mutations/supplyNonEth/useSupplyNonEth';

export { default as supplyEth } from './mutations/supplyEth';
export * from './mutations/supplyEth';
export { default as useSupplyEth } from './mutations/supplyEth/useSupplyEth';
export * from './mutations/supplyEth/useSupplyEth';

export { default as supplyNFT } from './mutations/supplyNFT';
export * from './mutations/supplyNFT';
export { default as useSupplyNFT } from './mutations/supplyNFT/useSupplyNFT';
export * from './mutations/supplyNFT/useSupplyNFT';

export { default as redeem } from './mutations/redeem';
export * from './mutations/redeem';
export { default as useRedeem } from './mutations/redeem/useRedeem';

export { default as redeemNFT } from './mutations/redeemNFT';
export * from './mutations/redeemNFT';
export { default as useRedeemNFT } from './mutations/redeemNFT/useRedeemNFT';

export { default as repayNonEthOToken } from './mutations/repayNonEthOToken';
export * from './mutations/repayNonEthOToken';
export { default as useRepayNonEthOToken } from './mutations/repayNonEthOToken/useRepayNonEthOToken';

export { default as repayEth } from './mutations/repayEth';
export * from './mutations/repayEth';
export { default as useRepayEth } from './mutations/repayEth/useRepayEth';

export { default as redeemUnderlying } from './mutations/redeemUnderlying';
export * from './mutations/redeemUnderlying';
export { default as useRedeemUnderlying } from './mutations/redeemUnderlying/useRedeemUnderlying';

export { default as claimXcnReward } from './mutations/claimXcnReward';
export * from './mutations/claimXcnReward';
export { default as useClaimXcnReward } from './mutations/claimXcnReward/useClaimXcnReward';

export { default as borrowOToken } from './mutations/borrowOToken';
export * from './mutations/borrowOToken';
export { default as useBorrowOToken } from './mutations/borrowOToken/useBorrowOToken';

export { default as useRepayOToken } from './mutations/useRepayOToken';

export { default as createProposal } from './mutations/createProposal';
export * from './mutations/createProposal';
export { default as useCreateProposal } from './mutations/createProposal/useCreateProposal';

export { default as cancelProposal } from './mutations/cancelProposal';
export * from './mutations/cancelProposal';
export { default as useCancelProposal } from './mutations/cancelProposal/useCancelProposal';

export { default as executeProposal } from './mutations/executeProposal';
export * from './mutations/executeProposal';
export { default as useExecuteProposal } from './mutations/executeProposal/useExecuteProposal';

export { default as queueProposal } from './mutations/queueProposal';
export * from './mutations/queueProposal';
export { default as useQueueProposal } from './mutations/queueProposal/useQueueProposal';

export { default as castVote } from './mutations/vote/castVote';
export * from './mutations/vote/castVote';
export { default as useCastVote } from './mutations/vote/useCastVote';

export { default as registerProxy } from './mutations/registerProxy';
export * from './mutations/registerProxy';
export { default as useRegisterProxy } from './mutations/registerProxy/useRegisterProxy';

export { default as depositPunk } from './mutations/depositPunk';
export * from './mutations/depositPunk';
export { default as useDepositPunk } from './mutations/depositPunk/useDepositPunk';

export { default as mintWPunks } from './mutations/mintWPunks';
export * from './mutations/mintWPunks';
export { default as useMintWPunks } from './mutations/mintWPunks/useMintWPunks';

export { default as burnWPunks } from './mutations/burnWPunks';
export * from './mutations/burnWPunks';
export { default as useBurnWPunks } from './mutations/burnWPunks/useBurnWPunks';

export { default as swapTokens } from './mutations/swapTokens';
export * from './mutations/swapTokens';
export { default as useSwapTokens } from './mutations/swapTokens/useSwapTokens';

// Queries
export { default as getAssetsInAccount } from './queries/getAssetsInAccount';
export * from './queries/getAssetsInAccount';
export { default as useGetAssetsInAccount } from './queries/getAssetsInAccount/useGetAssetsInAccount';

export { default as getHypotheticalAccountLiquidity } from './queries/getHypotheticalAccountLiquidity';
export * from './queries/getHypotheticalAccountLiquidity';

export { default as getMarkets } from './queries/getMarkets';
export * from './queries/getMarkets';
export { default as useGetMarkets } from './queries/getMarkets/useGetMarkets';

export { default as getOTokenBalancesAll } from './queries/getOTokenBalancesAll';
export * from './queries/getOTokenBalancesAll';
export { default as useGetOTokenBalancesAll } from './queries/getOTokenBalancesAll/useGetOTokenBalancesAll';

export { default as getOTokenBalanceOf } from './queries/getOTokenBalanceOf';
export * from './queries/getOTokenBalanceOf';
export { default as useGetOTokenBalanceOf } from './queries/getOTokenBalanceOf/useGetOTokenBalanceOf';

export { default as getXcnReward } from './queries/getXcnReward';
export * from './queries/getXcnReward';
export { default as useGetXcnReward } from './queries/getXcnReward/useGetXcnReward';

export { default as getAllowance } from './queries/getAllowance';
export * from './queries/getAllowance';
export { default as useGetAllowance } from './queries/getAllowance/useGetAllowance';

export { default as getIsApprovedForAll } from './queries/getIsApprovedForAll';
export * from './queries/getIsApprovedForAll';
export { default as useGetIsApprovedForAll } from './queries/getIsApprovedForAll/useGetIsApprovedForAll';

export { default as getBalanceOf } from './queries/getBalanceOf';
export * from './queries/getBalanceOf';
export { default as useGetBalanceOf } from './queries/getBalanceOf/useGetBalanceOf';

export { default as getTokenBalances } from './queries/getTokenBalances';
export * from './queries/getTokenBalances';
export { default as useGetTokenBalances } from './queries/getTokenBalances/useGetTokenBalances';

export { default as useGetUserMarketInfo } from './queries/useGetUserMarketInfo';

export { default as useGetTreasuryTotals } from './queries/useGetTreasuryTotals';

export { default as getMarketHistory } from './queries/getMarketHistory';
export * from './queries/getMarketHistory';
export { default as useGetMarketHistory } from './queries/getMarketHistory/useGetMarketHistory';

export { default as getOTokenCash } from './queries/getOTokenCash';
export * from './queries/getOTokenCash';
export { default as useGetOTokenCash } from './queries/getOTokenCash/useGetOTokenCash';

export { default as getOTokenInterestRateModel } from './queries/getOTokenInterestRateModel';
export * from './queries/getOTokenInterestRateModel';
export { default as useGetOTokenInterestRateModel } from './queries/getOTokenInterestRateModel/useGetOTokenInterestRateModel';

export { default as getOTokenApySimulations } from './queries/getOTokenApySimulations';
export * from './queries/getOTokenApySimulations';
export { default as useGetOTokenApySimulations } from './queries/getOTokenApySimulations/useGetOTokenApySimulations';

export { default as getCurrentVotes } from './queries/getCurrentVotes';
export * from './queries/getCurrentVotes';
export { default as useGetCurrentVotes } from './queries/getCurrentVotes/useGetCurrentVotes';

export { default as getStakingInfos } from './queries/getStakingInfos';
export * from './queries/getStakingInfos';
export { default as useGetStakingInfos } from './queries/getStakingInfos/useGetStakingInfos';

export { default as getStakingApy } from './queries/getStakingApy';
export * from './queries/getStakingApy';
export { default as useGetStakingApy } from './queries/getStakingApy/useGetStakingApy';

export { default as getTransactions } from './queries/getTransactions';
export * from './queries/getTransactions';
export { default as useGetTransactions } from './queries/getTransactions/useGetTransactions';

export { default as getDailyXcn } from './queries/getDailyXcn';
export * from './queries/getDailyXcn';
export { default as useGetDailyXcn } from './queries/getDailyXcn/useGetDailyXcn';

export { default as getProposals } from './queries/getProposals';
export * from './queries/getProposals';
export { default as useGetProposals } from './queries/getProposals/useGetProposals';

export * from './queries/getStakeHistories';
export { default as useGetStakeHistories } from './queries/getStakeHistories/useGetStakeHistories';

export { default as getProposal } from './queries/getProposals/getProposal';
export * from './queries/getProposals/getProposal';
export { default as useGetProposal } from './queries/getProposals/useGetProposal';

export { default as getVoteReceipt } from './queries/getVoteReceipt';
export * from './queries/getVoteReceipt';
export { default as useGetVoteReceipt } from './queries/getVoteReceipt/useGetVoteReceipt';

export { default as getVoters } from './queries/getVoters';
export * from './queries/getVoters';
export { default as useGetVoters } from './queries/getVoters/useGetVoters';

export { default as getVoterDetails } from './queries/getVoterDetails';
export * from './queries/getVoterDetails';
export { default as useGetVoterDetails } from './queries/getVoterDetails/useGetVoterDetails';

export { default as getVoterHistory } from './queries/getVoterHistory';
export * from './queries/getVoterHistory';
export { default as useGetVoterHistory } from './queries/getVoterHistory/useGetVoterHistory';

export { default as getVoterAccounts } from './queries/getVoterAccounts';
export * from './queries/getVoterAccounts';
export { default as useGetVoterAccounts } from './queries/getVoterAccounts/useGetVoterAccounts';

export { default as getProposalThreshold } from './queries/getProposalThreshold';
export * from './queries/getProposalThreshold';
export { default as useGetProposalThreshold } from './queries/getProposalThreshold/useGetProposalThreshold';

export { default as getProposalState } from './queries/getProposalState';
export * from './queries/getProposalState';
export { default as useGetProposalState } from './queries/getProposalState/useGetProposalState';

export { default as getLatestProposalIdByProposer } from './queries/getLatestProposalIdByProposer';
export * from './queries/getLatestProposalIdByProposer';
export { default as useGetLatestProposalIdByProposer } from './queries/getLatestProposalIdByProposer/useGetLatestProposalIdByProposer';

export { default as getBlockNumber } from './queries/getBlockNumber';
export * from './queries/getBlockNumber';
export { default as useGetBlockNumber } from './queries/getBlockNumber/useGetBlockNumber';

export { default as getProposalEta } from './queries/getProposalEta';
export * from './queries/getProposalEta';
export { default as useGetProposalEta } from './queries/getProposalEta/useGetProposalEta';

export { default as getProxies } from './queries/getProxies';
export * from './queries/getProxies';
export { default as useGetProxies } from './queries/getProxies/useGetProxies';

export { default as getOwnedPunkIds } from './queries/getOwnedPunkIds';
export * from './queries/getOwnedPunkIds';
export { default as useGetOwnedPunkIds } from './queries/getOwnedPunkIds/useGetOwnedPunkIds';

export { default as getOwnedWPunksIds } from './queries/getOwnedWPunksIds';
export * from './queries/getOwnedWPunksIds';
export { default as useGetOwnedWPunksIds } from './queries/getOwnedWPunksIds/useGetOwnedWPunksIds';

export { default as liquidateWithSingleRepay } from './mutations/liquidateWithSingleRepay';
export * from './mutations/liquidateWithSingleRepay';
export { default as useLiquidateWithSingleRepay } from './mutations/liquidateWithSingleRepay/useLiquidateWithSingleRepay';

export { default as liquidateBorrow } from './mutations/liquidateBorrow';
export * from './mutations/liquidateBorrow';
export { default as useLiquidateBorrow } from './mutations/liquidateBorrow/useLiquidateBorrow';

export { default as stakeXcn } from './mutations/stakeXcn';
export * from './mutations/stakeXcn';
export { default as useStakeXcn } from './mutations/stakeXcn/useStakeXcn';

export { default as withdrawXcn } from './mutations/withdrawXcn';
export * from './mutations/withdrawXcn';
export { default as useWithdrawXcn } from './mutations/withdrawXcn/useWithdrawXcn';

export { default as claimXcn } from './mutations/claimXcn';
export * from './mutations/claimXcn';
export { default as useClaimXcn } from './mutations/claimXcn/useClaimXcn';

export { default as getUniSwapPairs } from './queries/getUniSwapPairs';
export * from './queries/getUniSwapPairs';
export { default as useGetUniSwapPairs } from './queries/getUniSwapPairs/useGetUniSwapPairs';
