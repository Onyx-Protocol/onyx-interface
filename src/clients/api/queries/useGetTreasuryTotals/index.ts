import BigNumber from 'bignumber.js';
import { useMemo } from 'react';
import { Market } from 'types';

import { useGetMarkets } from 'clients/api';

export interface Data {
  treasuryTotalSupplyBalanceCents: BigNumber;
  treasuryTotalBorrowBalanceCents: BigNumber;
  treasuryTotalBalanceCents: BigNumber;
  treasuryTotalAvailableLiquidityBalanceCents: BigNumber;
}

export interface UseGetTreasuryTotalsOutput {
  isLoading: boolean;
  data: Data;
}

const useGetTreasuryTotals = (): UseGetTreasuryTotalsOutput => {
  const {
    data: getMarketsData = {
      markets: [] as Market[],
    },
    isLoading: isGetMarketsLoading,
  } = useGetMarkets({
    placeholderData: {
      markets: [],
      dailyXcnWei: new BigNumber(0),
    },
  });

  const { markets } = getMarketsData;
  const {
    treasuryTotalSupplyBalanceCents,
    treasuryTotalBorrowBalanceCents,
    treasuryTotalBalanceCents,
    treasuryTotalAvailableLiquidityBalanceCents,
  } = useMemo(() => {
    const data = markets.reduce(
      (acc, curr) => {
        acc.treasuryTotalBalanceCents = acc.treasuryTotalBalanceCents.plus(
          new BigNumber(curr.totalReserves)
            .dividedBy(new BigNumber(10).pow(curr.underlyingDecimal))
            .multipliedBy(curr.tokenPrice)
            .times(100),
        );

        acc.treasuryTotalSupplyBalanceCents = acc.treasuryTotalSupplyBalanceCents.plus(
          curr.treasuryTotalSupplyCents,
        );

        acc.treasuryTotalBorrowBalanceCents = acc.treasuryTotalBorrowBalanceCents.plus(
          curr.treasuryTotalBorrowsCents,
        );

        acc.treasuryTotalAvailableLiquidityBalanceCents =
          acc.treasuryTotalAvailableLiquidityBalanceCents.plus(curr.liquidity.times(100));

        return acc;
      },
      {
        treasuryTotalSupplyBalanceCents: new BigNumber(0),
        treasuryTotalBorrowBalanceCents: new BigNumber(0),
        treasuryTotalBalanceCents: new BigNumber(0),
        treasuryTotalAvailableLiquidityBalanceCents: new BigNumber(0),
      },
    );
    return data;
  }, [markets]);

  return {
    data: {
      treasuryTotalSupplyBalanceCents,
      treasuryTotalBorrowBalanceCents,
      treasuryTotalBalanceCents,
      treasuryTotalAvailableLiquidityBalanceCents,
    },
    isLoading: isGetMarketsLoading,
  };
};

export default useGetTreasuryTotals;
