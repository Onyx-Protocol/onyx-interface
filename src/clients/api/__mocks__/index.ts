import BigNumber from 'bignumber.js';
import { MutationObserverOptions, useMutation, useQuery } from 'react-query';

import fakeAddress from '__mocks__/models/address';
import { assetData } from '__mocks__/models/asset';
import proposals from '__mocks__/models/proposals';
import transactionReceipt from '__mocks__/models/transactionReceipt';
import voters from '__mocks__/models/voters';
import FunctionKey from 'constants/functionKey';

import { GetBalanceOfInput } from '../queries/getBalanceOf';

// Queries
export const getBlockNumber = jest.fn();
export const useGetBlockNumber = () => useQuery(FunctionKey.GET_BLOCK_NUMBER, getBlockNumber);

export const getAssetsInAccount = jest.fn();
export const useGetAssetsInAccount = () =>
  useQuery(FunctionKey.GET_ASSETS_IN_ACCOUNT, getAssetsInAccount);

export const getHypotheticalAccountLiquidity = jest.fn();

export const getMarkets = jest.fn();
export const useGetMarkets = () => useQuery(FunctionKey.GET_MARKETS, getMarkets);

export const getMarketHistory = jest.fn();
export const useGetMarketHistory = () => useQuery(FunctionKey.GET_MARKET_HISTORY, getMarketHistory);

export const getOTokenBalancesAll = jest.fn();
export const useGetOTokenBalancesAll = jest.fn(() =>
  useQuery(FunctionKey.GET_O_TOKEN_BALANCES_ALL, getOTokenBalancesAll),
);

export const getXcnReward = jest.fn();
export const useGetXcnReward = () => useQuery(FunctionKey.GET_XCN_REWARD, getXcnReward);

export const getOTokenBalanceOf = jest.fn();
export const useGetOTokenBalanceOf = () =>
  useQuery(FunctionKey.GET_O_TOKEN_BALANCE, getOTokenBalanceOf);

export const getOTokenBorrowBalance = jest.fn();
export const useGetOTokenBorrowBalance = () =>
  useQuery(FunctionKey.GET_O_TOKEN_BORROW_BALANCE, getOTokenBorrowBalance);

export const getAllowance = jest.fn();
export const useGetAllowance = () => useQuery(FunctionKey.GET_TOKEN_ALLOWANCE, getAllowance);

export const getBalanceOf = jest.fn();
export const useGetBalanceOf = (input: Omit<GetBalanceOfInput, 'web3'>) =>
  useQuery(
    [
      FunctionKey.GET_BALANCE_OF,
      {
        accountAddress: input.accountAddress,
        tokenAddress: input.token.address,
      },
    ],
    () => getBalanceOf(input),
  );

export const getTokenBalances = jest.fn();
export const useGetTokenBalances = () => useQuery(FunctionKey.GET_TOKEN_BALANCES, getTokenBalances);

export const getXcnWithdrawableAmount = jest.fn();
export const useGetXcnWithdrawableAmount = () =>
  useQuery(FunctionKey.GET_XCN_WITHDRAWABLE_AMOUNT, getXcnWithdrawableAmount);

export const getOTokenCash = jest.fn();
export const useGetOTokenCash = () => useQuery(FunctionKey.GET_O_TOKEN_CASH, getOTokenCash);

export const getOTokenInterestRateModel = jest.fn();
export const useGetOTokenInterestRateModel = () =>
  useQuery(FunctionKey.GET_O_TOKEN_INTEREST_RATE_MODEL, getOTokenInterestRateModel);

export const getOTokenApySimulations = jest.fn();
export const useGetOTokenApySimulations = () =>
  useQuery(FunctionKey.GET_O_TOKEN_APY_SIMULATIONS, getOTokenApySimulations);

export const getOTokenSupplyRate = jest.fn();

export const getOTokenBorrowRate = jest.fn();

export const getTransactions = jest.fn();
export const useGetTransactions = jest.fn(() =>
  useQuery([FunctionKey.GET_TRANSACTIONS, {}], getTransactions),
);

export const useGetTreasuryTotals = jest.fn();

export const useGetUserMarketInfo = jest.fn(() => ({
  isLoading: false,
  data: {
    assets: assetData,
  },
}));

export const getCurrentVotes = jest.fn(() => new BigNumber(100000000000000000));
export const useGetCurrentVotes = () => useQuery(FunctionKey.GET_CURRENT_VOTES, getCurrentVotes);

export const getPriorVotes = jest.fn(() => new BigNumber(100000000000000000));
export const useGetPriorVotes = () => useQuery(FunctionKey.GET_PRIOR_VOTES, getPriorVotes);

export const getProposals = jest.fn();
export const useGetProposals = () => useQuery(FunctionKey.GET_PROPOSALS, getProposals);

export const getProposal = jest.fn(async () => (await proposals)[0]);
export const useGetProposal = () => useQuery(FunctionKey.GET_PROPOSAL, getProposal);

export const getDailyXcn = jest.fn();
export const useGetDailyXcn = () => useQuery(FunctionKey.GET_O_TOKEN_DAILY_XCN, getDailyXcn);

export const getVoters = jest.fn(() => voters);
export const useGetVoters = jest.fn(() => useQuery(FunctionKey.GET_VOTERS, getVoters));

export const getVoterHistory = jest.fn();
export const useGetVoterHistory = () => useQuery(FunctionKey.GET_VOTER_HISTORY, getVoterHistory);

export const getVoterDetails = jest.fn();
export const useGetVoterDetails = () => useQuery(FunctionKey.GET_VOTER_DETAILS, getVoterDetails);

export const getVoteReceipt = jest.fn();
export const useGetVoteReceipt = () => useQuery(FunctionKey.GET_VOTE_RECEIPT, getVoteReceipt);

export const getVoteDelegateAddress = jest.fn();
export const useGetVoteDelegateAddress = () =>
  useQuery([FunctionKey.GET_VOTE_DELEGATE_ADDRESS, fakeAddress], getVoteDelegateAddress);

export const getLatestProposalIdByProposer = jest.fn();
export const useGetLatestProposalIdByProposer = () =>
  useQuery(
    [FunctionKey.GET_LATEST_PROPOSAL_ID_BY_PROPOSER, fakeAddress],
    getLatestProposalIdByProposer,
  );
export const useGetActiveProposal = jest.fn();

export const getVoterAccounts = jest.fn();
export const useGetVoterAccounts = () => useQuery(FunctionKey.GET_VOTER_ACCOUNTS, getVoterAccounts);

export const getProposalThreshold = jest.fn(() => new BigNumber('10000000000000000000000'));
export const useGetProposalThreshold = () =>
  useQuery(FunctionKey.GET_PROPOSAL_THRESHOLD, getProposalThreshold);

export const getProposalState = jest.fn();
export const useGetProposalState = () => useQuery(FunctionKey.GET_PROPOSAL_STATE, getProposalState);

export const getProposalEta = jest.fn();
export const useGetProposalEta = () => useQuery(FunctionKey.GET_PROPOSAL_ETA, getProposalEta);

// Mutations
export const approveToken = jest.fn();
export const useApproveToken = (options?: MutationObserverOptions) =>
  useMutation(FunctionKey.APPROVE_TOKEN, approveToken, options);

export const enterMarkets = jest.fn();
export const useEnterMarkets = (options?: MutationObserverOptions) =>
  useMutation(FunctionKey.ENTER_MARKETS, enterMarkets, options);

export const exitMarket = jest.fn();
export const useExitMarket = (options?: MutationObserverOptions) =>
  useMutation(FunctionKey.EXIT_MARKET, exitMarket, options);

export const claimXcnReward = jest.fn();
export const useClaimXcnReward = (options?: MutationObserverOptions) =>
  useMutation(FunctionKey.CLAIM_XCN_REWARD, claimXcnReward, options);

export const repayEth = jest.fn();
export const useRepayEth = () => useMutation(FunctionKey.REPAY_ETH, repayEth);

export const repayNonEthOToken = jest.fn();
export const useRepayNonEthOToken = () =>
  useMutation(FunctionKey.REPAY_NON_ETH_O_TOKEN, repayNonEthOToken);

export const useRepayOToken = useRepayNonEthOToken;

export const supply = jest.fn();
export const useSupply = () => useMutation(FunctionKey.SUPPLY, supply);
export const supplyNonEth = jest.fn();
export const useSupplyNonEth = () => useMutation(FunctionKey.SUPPLY, supplyNonEth);

export const supplyEth = jest.fn();
export const useSupplyEth = () => useMutation(FunctionKey.SUPPLY_ETH, supplyEth);

export const redeem = jest.fn();
export const useRedeem = () => useMutation(FunctionKey.REDEEM, redeem);

export const redeemUnderlying = jest.fn();
export const useRedeemUnderlying = () => useMutation(FunctionKey.REDEEM, redeemUnderlying);

export const borrowOToken = jest.fn();
export const useBorrowOToken = () => useMutation(FunctionKey.BORROW_O_TOKEN, borrowOToken);

export const setVoteDelegate = jest.fn();
export const useSetVoteDelegate = (options?: MutationObserverOptions) =>
  useMutation(FunctionKey.SET_VOTE_DELEGATE, setVoteDelegate, options);

export const createProposal = jest.fn();
export const useCreateProposal = (options?: MutationObserverOptions) =>
  useMutation(FunctionKey.CREATE_PROPOSAL, createProposal, options);

export const cancelProposal = jest.fn(async () => transactionReceipt);
export const useCancelProposal = (options?: MutationObserverOptions) =>
  useMutation(FunctionKey.CANCEL_PROPOSAL, cancelProposal, options);

export const executeProposal = jest.fn(async () => transactionReceipt);
export const useExecuteProposal = (options?: MutationObserverOptions) =>
  useMutation(FunctionKey.EXECUTE_PROPOSAL, executeProposal, options);

export const queueProposal = jest.fn(async () => transactionReceipt);
export const useQueueProposal = (options?: MutationObserverOptions) =>
  useMutation(FunctionKey.QUEUE_PROPOSAL, queueProposal, options);

export const castVote = jest.fn();
export const useCastVote = (options?: MutationObserverOptions) =>
  useMutation(FunctionKey.CAST_VOTE, castVote, options);

export const swapTokens = jest.fn();
export const useSwapTokens = (options?: MutationObserverOptions) =>
  useMutation(FunctionKey.SWAP_TOKENS, swapTokens, options);

export const getIsApprovedForAll = jest.fn();
export const useGetIsApprovedForAll = (options?: MutationObserverOptions) =>
  useMutation(FunctionKey.GET_NFT_ALLOWANCE, getIsApprovedForAll, options);

export const approveNftForAll = jest.fn();
export const useApproveNftForAll = (options?: MutationObserverOptions) =>
  useMutation(FunctionKey.APPROVE_NFT_TOKEN, approveNftForAll, options);
