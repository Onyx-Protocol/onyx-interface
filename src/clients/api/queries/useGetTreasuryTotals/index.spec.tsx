import React from 'react';

import { markets } from '__mocks__/models/markets';
import { oTokenBalanceTreasury } from '__mocks__/models/oTokenBalanceTreasury';
import { getMarkets, useGetOTokenBalancesAll } from 'clients/api';
import renderComponent from 'testUtils/renderComponent';

import useGetTreasuryTotals, { UseGetTreasuryTotalsOutput } from '.';

jest.mock('clients/api');

describe('api/queries/useGetTreasuryTotals', () => {
  beforeEach(() => {
    (getMarkets as jest.Mock).mockImplementation(() => ({ markets }));

    (useGetOTokenBalancesAll as jest.Mock).mockImplementation(() => ({
      data: {
        balances: oTokenBalanceTreasury,
      },
    }));
  });

  it('calculates totals correctly', async () => {
    let data: UseGetTreasuryTotalsOutput['data'];

    const CallMarketContext = () => {
      ({ data } = useGetTreasuryTotals());
      expect(data).toMatchSnapshot();
      return <div />;
    };

    renderComponent(<CallMarketContext />);
  });
});
