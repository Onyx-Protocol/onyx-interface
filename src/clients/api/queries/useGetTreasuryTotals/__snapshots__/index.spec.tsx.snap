// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api/queries/useGetTreasuryTotals calculates totals correctly 1`] = `
Object {
  "treasuryTotalAvailableLiquidityBalanceCents": "0",
  "treasuryTotalBalanceCents": "0",
  "treasuryTotalBorrowBalanceCents": "0",
  "treasuryTotalSupplyBalanceCents": "0",
}
`;

exports[`api/queries/useGetTreasuryTotals calculates totals correctly 2`] = `
Object {
  "treasuryTotalAvailableLiquidityBalanceCents": "1.009809011837545743841842745570829318278501e+26",
  "treasuryTotalBalanceCents": "4725237471469.07794503682728247608883",
  "treasuryTotalBorrowBalanceCents": "96086700086.7474427810189748",
  "treasuryTotalSupplyBalanceCents": "1.009799010007167652859482000681936356943184e+26",
}
`;
